ast2vec/repo2/base.py

```diff
@@ -126,22 +127,41 @@ def thread_loop(thread_index):
                             break
                         try:
                             folder, filename, language = task
-                            filepath = os.path.join(folder, filename).encode('utf8')
+                            filepath_naive = os.path.join(folder, filename).encode('utf8')
                             # I need .encode('utf8') to avoid problems
                             # with bad symbols in file names on Ubuntu
 
                             # Check if file path is symlink
-                            if os.path.islink(filepath):
-                                filepath = os.readlink(filepath)
+                            islink = os.path.islink(filepath_naive)
+                            if islink:
+                                filepath = os.readlink(filepath_naive)
+                            else:
+                                filepath = filepath_naive
+                            if not os.path.exists(filepath):
+                                self._log.warning('File %s does not exist.\n\t%s',
+                                                  filepath.decode('utf8'),
+                                                  'Get it from %s symlink.' %
+                                                  filepath_naive.decode('utf8')
+                                                  if islink else 'And it is not a symlink...')
+                                queue_out.put_nowait(None)
+                                continue
 
                             size = os.stat(filepath).st_size
                             if size > self.MAX_FILE_SIZE:
-                                self._log.warning("%s is too big - %d", filepath, size)
+                                self._log.warning("%s is too big - %d",
+                                                  filepath.decode('utf8'), size)
                                 queue_out.put_nowait(None)
                                 continue
+                            try:
+                                response = self._bblfsh[thread_index].parse(
+                                    filepath, language=language, timeout=self._timeout)
+                            except DecodeError as e:
+                                self._log.warning("bblfsh raise an DecodeError exception. "
+                                                  "Usually it is because protobuf package v3.3.0. "
```

Comment:
Probably your protobuf is >= v3.3.0 and you hit https://github.com/bblfsh/server/issues/59#issuecomment-318125752